drivers/dac/dac_dacx3608.c

```diff
@@ -0,0 +1,282 @@
+				uint32_t value)
+{
+	const struct dacx3608_config *config = dev->config;
+	struct dacx3608_data *data = dev->data;
+	uint16_t regval;
+	int ret;
+
+	if (channel > DACX3608_MAX_CHANNEL - 1) {
+		LOG_ERR("Unsupported channel %d", channel);
+		return -ENOTSUP;
+	}
+
+	if (!(data->configured & BIT(channel))) {
+		LOG_ERR("Channel %d not initialized", channel);
+		return -EINVAL;
+	}
+
+	if (value >= (1 << (config->resolution))) {
+		LOG_ERR("Value %d out of range", value);
+		return -EINVAL;
+	}
+
+	/*
+	 * Shift passed value two times left because first two bits are Don't Care
+	 *
+	 * DACn_DATA register format:
+	 *
+	 * | 15 14 13 12 |      11 10 9 8 7 6 5 4 3 2      |    1 0     |
+	 * |-------------|---------------------------------|------------|
+	 * | Don't Care  |  DAC53608[9:0] / DAC43608[7:0]  | Don't Care |
+	 */
+	regval = value << 2;
+	regval &= 0xFFFF;
+
+	ret = dacx3608_reg_write(dev, DACX3608_REG_DACA_DATA + channel, regval);
+	if (ret) {
+		LOG_ERR("Unable to set value %d on channel %d", value, channel);
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int dacx3608_soft_reset(const struct device *dev)
+{
+	uint16_t regval = DACX3608_SW_RST;
+	int ret;
+
+	ret = dacx3608_reg_write(dev, DACX3608_REG_STATUS_TRIGGER, regval);
+	if (ret) {
+		return -EIO;
+	}
+	k_msleep(DACX3608_POR_DELAY);
+
+	return 0;
+}
+
+static int dacx3608_device_id_check(const struct device *dev)
+{
+	struct dacx3608_data *data = dev->data;
+	int ret;
+
+	ret = dacx3608_reg_read(dev, DACX3608_REG_STATUS_TRIGGER, &data->dev_id);
+	if (ret) {
+		LOG_ERR("Unable to read device ID");
+		return -EIO;
+	}
+
+	switch (data->dev_id) {
+	case DAC43608_DEVICE_ID:
+	case DAC53608_DEVICE_ID:
+		LOG_DBG("Device ID %#4x", data->dev_id);
+		break;
+	default:
+		LOG_ERR("Unknown Device ID %#4x", data->dev_id);
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int dacx3608_init(const struct device *dev)
+{
+	const struct dacx3608_config *config = dev->config;
+	struct dacx3608_data *data = dev->data;
+	int ret;
+
+	data->i2c = device_get_binding(config->i2c_bus);
+	if (!data->i2c) {
+		LOG_ERR("Could not find I2C device");
+		return -EINVAL;
+	}
+
+	ret = dacx3608_soft_reset(dev);
+	if (ret) {
+		LOG_ERR("Soft-reset failed");
+		return ret;
+	}
+
+	ret = dacx3608_device_id_check(dev);
+	if (ret) {
+		return ret;
+	}
+
+	data->configured = 0;
+
+	LOG_DBG("Init complete");
+
+	return 0;
+}
+
+static const struct dac_driver_api dacx3608_driver_api = {
+	.channel_setup = dacx3608_channel_setup,
+	.write_value = dacx3608_write_value,
+};
+
+#define INST_DT_DACX3608(inst, t) DT_INST(inst, ti_dac##t)
+
+#define DACX3608_DEVICE(t, n, res) \
+	static struct dacx3608_data dac##t##_data_##n; \
+	static const struct dacx3608_config dac##t##_config_##n = { \
+		.i2c_bus = DT_BUS_LABEL(INST_DT_DACX3608(n, t)), \
+		.i2c_addr = DT_REG_ADDR(INST_DT_DACX3608(n, t)), \
+		.resolution = res, \
+	}; \
+	DEVICE_DT_DEFINE(INST_DT_DACX3608(n, t), \
```

Comment:
Great that you have already updated to the new DT define macros merged in between :+1:

Reply:
I noticed that commit so I already adapted this :smile: